### ISSUE TITLE
Bug of the same note shown multiple times

### ISSUE THREAD

**Issue by nuttyartist** [nuttyartist/notes#175]
There is a serious bug in version 1.0.0 since moving to the new database architecture. What happens is that sometimes after closing the app and re-running it again, some notes get duplicated. Sometimes the duplications are an early version of the note, sometimes it's the exact note, etc... 

The good thing is that the original note is always there and nothing gets deleted, just some notes get duplicated in a weird way. While I'm having this bug for a long time, I still didn't find a way to replicate it in a consistent manner.

I attached some screenshots I gathered to show the bug in action.

![screenshot from 2017-04-29 14-28-20](https://user-images.githubusercontent.com/16375940/30074700-245fdb04-927b-11e7-9f11-85c5370497ba.png)
![screenshot from 2017-05-04 22-03-24](https://user-images.githubusercontent.com/16375940/30074740-44c28b76-927b-11e7-9cf0-a64d65cde730.png)
![selection_003](https://user-images.githubusercontent.com/16375940/30074756-50102fd8-927b-11e7-81b0-e3d072e4bd3f.png)

**Comment by theshadowx** [comment]
Could you give me more details about the OS and could you show me a screenshot of the directory where there are the settings, please

**Comment by nuttyartist** [comment]
It happened for me both on Linux and currently, Mac. I think it might be a cross-platform problem.
Don't you ever experience that bug?

Also, I don't think the Settings.ini is related but this is its content:

```
[General]
dontShowUpdateWindow=false
splitterSizes=@ByteArray(\0\0\0\xff\0\0\0\x1\0\0\0\x2\0\0\0\xbb\0\0\x2s\x1\0\0\0\0\x1\0\0\0\x1\0)
version=1.0.0
windowGeometry=@ByteArray(\x1\xd9\xd0\xcb\0\x2\0\0\0\0\x2\x15\0\0\0\x32\0\0\x5L\0\0\x2\x9e\0\0\x2\x15\0\0\0\x32\0\0\x5L\0\0\x2\x9e\0\0\0\0\0\0\0\0\x5\xa0)

```

**Comment by theshadowx** [comment]
for the screenshot I mean the content of the directory

**Comment by nuttyartist** [comment]
You mean the notes.db file of the screenshots? I took this screenshots a long time ago, I wanted to gather as many screenshots of this bug I could of notes that I can share.

**Comment by theshadowx** [comment]
I mean file browser, just to know the files that exist in this directory

**Comment by nuttyartist** [comment]
Oh, only notes.db and Settings.ini. This is how it always been.

**Comment by theshadowx** [comment]
ok, so there wasn't Notes.ini before ?

**Comment by nuttyartist** [comment]
There were. After the automatic conversion, I deleted it. Do you think it's related to that conversion? Even though it's been a long time ago and it's still happening on new notes?

**Comment by theshadowx** [comment]
ah ok

**Comment by nuttyartist** [comment]
This bug seems to be fixed after #179. I didn't encounter this bug once after the fix, where usually I will tackle into that quite frequently.
